jupyter_server/serverapp.py

```diff
@@ -1427,6 +1427,12 @@ def _default_allow_remote(self) -> bool:
                         """,
     )
 
+    allow_setup_custom_env_variables = Bool(
```

Comment:
A couple of comments on this new trait:

1. I think we should shorten the name if we can 😅 
2. Even though I said we should shorten it, I think we need the word `kernel` in the name somewhere 😆 . How about `accept_kernel_env_vars`? It's slightly shorter and more clear (IMO).
3. I think this trait needs to be passed as a tornado setting so that the request handlers can use this trait too.